src/module/documents/active-effect.mjs

```diff
@@ -16,4 +17,22 @@ export class DrawSteelActiveEffect extends ActiveEffect {
     super.prepareDerivedData();
     Hooks.callAll("ds.prepareActiveEffectData", this);
   }
+
+  /** @import {ActiveEffectDuration, EffectDurationData} from "../data/effect/_types" */
+
+  /**
+   * Compute derived data related to active effect duration.
+   * @returns {Omit<ActiveEffectDuration, keyof EffectDurationData>}
+   * @protected
+   */
+  _prepareDuration() {
+    return this.system._prepareDuration ?? super._prepareDuration();
+  }
+
+  /**
+   * Check if the effect's subtype has special handling, otherwise fallback to normal `duration` and `statuses` check
+   */
+  get isTemporary() {
+    return this.system.isTemporary ?? super.isTemporary;
+  }
```

Comment:
Since it does not rely solely on the system property, best not mark that as API. Whether an effect is temporary should still be derived from the document class (and whatever that calls):
```suggestion
  get isTemporary() {
    return this.system._isTemporary ?? super.isTemporary;
  }
```